tests/slostatusapi_test.go

```diff
@@ -0,0 +1,273 @@
+	t.Cleanup(func() { v1Delete(t, []manifest.Object{slo3, slo4, slo5}) })
+
+	limit := 2
+	firstResponse, err := tryExecuteGetSLOsV2Request(t, func() (v2.SLOListResponse, error) {
+		return client.SLOStatusAPI().V2().GetSLOs(ctx, v2.GetSLOsRequest{Limit: limit})
+	}, limit)
+	require.NoError(t, err)
+	assert.NotEmpty(t, firstResponse)
+	assert.NotEmpty(t, firstResponse.Links.Self, "expected first response's self link to be set")
+	assert.NotEmpty(t, firstResponse.Links.Next, "expected first response's next link to be set")
+	firstCursor := firstResponse.Links.Cursor
+	require.NotEmpty(t, firstCursor)
+
+	secondResponse, err := tryExecuteGetSLOsV2Request(t, func() (v2.SLOListResponse, error) {
+		return client.SLOStatusAPI().V2().GetSLOs(ctx, v2.GetSLOsRequest{Limit: limit, Cursor: firstCursor})
+	}, limit)
+	require.NoError(t, err)
+	assert.NotEmpty(t, secondResponse)
+	assert.NotEmpty(t, secondResponse.Links.Self, "expected second response's self link to be set")
+	assert.NotEmpty(t, secondResponse.Links.Next, "expected second response's next link to be set")
+	secondCursor := secondResponse.Links.Cursor
+	require.NotEmpty(t, secondCursor)
+	assert.NotEqual(t, firstCursor, secondCursor)
+	assert.NotEqual(t, firstResponse, secondResponse)
+}
+
+func setupSLOListTest(t *testing.T) []manifest.Object {
+	t.Helper()
+	project := generateV1alphaProject(t)
+	service := newV1alphaService(t, v1alphaService.Metadata{
+		Name:    generateName(),
+		Project: project.GetName(),
+	})
+
+	dataSourceType := v1alpha.Datadog
+	directs := filterSlice(v1alphaSLODependencyDirects(t), func(o manifest.Object) bool {
+		typ, _ := o.(v1alphaDirect.Direct).Spec.GetType()
+		return typ == dataSourceType
+	})
+	require.Len(t, directs, 1)
+	direct := directs[0].(v1alphaDirect.Direct)
+
+	slo := getExample[v1alphaSLO.SLO](t,
+		manifest.KindSLO,
+		func(example v1alphaExamples.Example) bool {
+			dsGetter, ok := example.(dataSourceTypeGetter)
+			return ok && dsGetter.GetDataSourceType() == dataSourceType
+		},
+	)
+	slo.Spec.AnomalyConfig = nil
+	slo.Metadata.Name = generateName()
+	slo.Metadata.Project = project.GetName()
+	slo.Spec.Indicator.MetricSource = v1alphaSLO.MetricSourceSpec{
+		Name:    direct.Metadata.Name,
+		Project: direct.Metadata.Project,
+		Kind:    manifest.KindDirect,
+	}
+	slo.Spec.AlertPolicies = nil
+	slo.Spec.Service = service.Metadata.Name
+	slo.Spec.Objectives[0].Name = "good"
+	return []manifest.Object{project, service, slo}
+}
+
+func tryExecuteGetSLORequest[T any](t *testing.T, reqFunc func() (T, error)) (T, error) {
+	t.Helper()
+	ticker := time.NewTicker(5 * time.Second)
+	timer := time.NewTimer(time.Minute)
+	defer ticker.Stop()
+	defer timer.Stop()
+	var (
+		response T
+		err      error
+	)
+	for {
+		select {
+		case <-ticker.C:
+			response, err = reqFunc()
+			if err == nil {
+				return response, nil
+			}
+		case <-timer.C:
+			t.Error("timeout")
+			return response, err
+		}
+	}
+}
+
+func tryExecuteGetSLOsV1Request(
```

Comment:
you don't need these 3 functions, just `tryExecuteGetSLORequest` is enough, you can assert this part of code:
```go
if len(response.Data) != limit {
	err = fmt.Errorf("expected %d SLOs, got %d", limit, len(response.Data))
}
```
in the `reqFunc` and since your function is generic it can handle both `v1.SLOListResponse` and `v2.SLOListResponse`

Reply:
You're right, thanks! It's a neat solution.